torch/_inductor/lowering.py

```diff
@@ -1074,9 +1074,8 @@ def should_lower_cat_input(x) -> bool:
     if len(inputs) <= config.max_pointwise_cat_inputs:
         pointwise_uses = all(is_pointwise_use(use) for use in V.current_node.users)
         all_pointwise_inputs = all(should_lower_cat_input(inp) for inp in inputs)
-        any_pointwise_inputs = any(should_lower_cat_input(inp) for inp in inputs)
 
-        if all_pointwise_inputs or (any_pointwise_inputs and pointwise_uses):
+        if all_pointwise_inputs and pointwise_uses:
```

Comment:
I think we should check if the pointwise input will get fused into the reduction, instead of this change. if you had `out = torch.cat([torch.softmax(a, dim=-1) + 1, torch.softmax(b, dim=-1) + 1])` this would fuse

Reply:
I think it's not that easy for us at this point to predict whether the pointwise will get fused into a reduction.

Nevertheless, if there are no downstream pointwise ops to fuse into, then this is not beneficial no? So given that, I think we should err on the side of caution.

Reply:
I think we should just get rid of the `all_pointwise_inputs` heuristic. If there are `any_pointwise_inputs and pointwise_uses` then this saved global memory reads. 

If we need a more thorough fix for not fusing in reductions, I can work on that.

Reply:
Oh, that's right. I'll change the heuristic.

Reply:
While you're add it, could you skip pointwise cat for cpu? There was regression bc of it, see : https://github.com/pytorch/pytorch/issues/112139.